### PR TITLE
Remove `maybe` convention for internal optionals

### PR DESCRIPTION
**Pull request by aboedo**
ripped off the band-aid. 

Removed the convention since: 
- it's not a common convention
- the compiler already contains information for whether an object is optional
- Hungarian notation is not the standard for Swift
- It can be daunting for newcomers
- It makes our `internal` and `public` code inconsistent
- It's hard to remember

### Requirements: 
- #975  ✅

[sc-12748]

### CODE REVIEW

**Comment by aboedo** [comment]
moving back to draft, clearly needs more thought 😅

**Review by NachoSoto — APPROVED** [review]
🩹❌😍

File: Purchases/Purchasing/PurchasesOrchestrator.swift [hunk 1]
```diff
@@ -27,16 +27,16 @@ class PurchasesOrchestrator {
     var finishTransactions: Bool { systemInfo.finishTransactions }
     var allowSharingAppStoreAccount: Bool {
         get {
-            return maybeAllowSharingAppStoreAccount ?? identityManager.currentUserIsAnonymous
+            return _allowSharingAppStoreAccount ?? identityManager.currentUserIsAnonymous
```
Comment (aboedo) [inline Purchases/Purchasing/PurchasesOrchestrator.swift:30]: I was really not sure what to go with for this one, open to ideas!

File: SwiftStyleGuide.swift [hunk 2]
```diff
@@ -164,12 +159,11 @@ private extension String {
 
 }
 
-// swiftlint:disable identifier_name
 // Use one line per let in a guard with multiple lets.
-let maybe🌮 = restaurant.order("🥗")
-let maybe🥤 = restaurant.order("☕️")
-guard let veggieTaco = maybe🌮,
-      let coffee = maybe🥤 else {
+let taco = restaurant.order("taco")
```
Comment (taquitos) [inline SwiftStyleGuide.swift:163]: WHERE DID THE EMOJIS GO??

File: SwiftStyleGuide.swift [hunk 3]
```diff
@@ -47,14 +47,9 @@ struct MyStruct {
...
-    // don't use `maybe` for public properties, since it might be confusing to developers
-    // unfamiliar with our naming convention
-    /// maybeVenus is used for ...
-    public var saturn: String?
 
-    // use `maybe` prefix for optionals
-    /// maybeVenus is used for ...
-    internal var maybeVenus: String?
+    /// saturn is used for ...
+    public var saturn: String?
```
Comment (aboedo) [inline SwiftStyleGuide.swift:52]: updated style guide accordingly

File: SwiftStyleGuide.swift (hunk 2 above)
Comment (aboedo) [reply SwiftStyleGuide.swift:163]: LOL I didn't wanna deal with swiftlint complaining about variable length, but I can bring them back

**Comment by aboedo** [comment]
okay, I believe this should be ready to go now. I checked the sample apps and backend integration tests and things are working.

**Comment by aboedo** [comment]
@NachoSoto I made quite a few changes since the first "ready for review", would you mind taking another look?

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 4]
```diff
@@ -48,25 +48,25 @@ class BackendIntegrationTests: XCTestCase {
...
-            maybeReceivedOfferings = offerings
+            receivedOfferings = offerings
         }
         expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
 
         expect(receivedError).to(beNil())
-        let receivedOfferings = try XCTUnwrap(maybeReceivedOfferings)
-        expect(receivedOfferings.all).toNot(beEmpty())
+        let unwrappedOfferings = try XCTUnwrap(receivedOfferings)
+        expect(unwrappedOfferings.all).toNot(beEmpty())
```
[the remark this answers is not in the document]
Comment (NachoSoto) [reply BackendIntegrationTests/BackendIntegrationTests.swift:61]: You can use the same name, it'll shadow the previous declaration.

File: Purchases/Purchasing/PurchasesOrchestrator.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Purchasing/PurchasesOrchestrator.swift:30]: This matches what Apple typically does.

File: SwiftStyleGuide.swift (hunk 2 above)
Comment (taquitos) [reply SwiftStyleGuide.swift:163]: It's all good 😄

**Comment by taquitos** [comment]
Could you do me a solid and not merge this until I merge https://github.com/RevenueCat/purchases-ios/pull/975
I'm soooo close and I've been rebasing since mid - nov (my own fault, I know)

**Comment by aboedo** [comment]
@taquitos absolutely no worries! I'll add it as a requirement in the description and rebase once #975 is merged

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 5]
```diff
@@ -236,34 +236,34 @@ class BackendIntegrationTests: XCTestCase {
...
-            maybeProductID = offerings?.current?.monthly?.storeProduct.productIdentifier
+            productID = offerings?.current?.monthly?.storeProduct.productIdentifier
             completionCalled = true
         }
         
         expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
         completionCalled = false
         
-        let productID = try XCTUnwrap(maybeProductID)
+        let unwrappedProductID = try XCTUnwrap(productID)
```
Comment (NachoSoto) [inline BackendIntegrationTests/BackendIntegrationTests.swift:251]: Ditto in all these. I think you can reuse the name.

Comment (NachoSoto) [inline BackendIntegrationTests/BackendIntegrationTests.swift:251]: Ditto in all these. I think you can reuse the name.

File: Purchases/Error Handling/BackendErrorCode.swift [hunk 6]
```diff
@@ -43,26 +43,26 @@ enum BackendErrorCode: Int, Error {
...
+        guard let codeInt = codeInt else {
             self = .unknownBackendError
             return
         }
 
         self = BackendErrorCode(rawValue: codeInt) ?? .unknownBackendError
     }
 
-    static func maybeInt(maybeCodeObject: Any?) -> Int? {
+    static func int(codeObject: Any?) -> Int? {
```
Comment (NachoSoto) [inline Purchases/Error Handling/BackendErrorCode.swift:59]: Maybe `extractCodeNumber(from:)`?

File: RevenueCat.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -2275,9 +2275,10 @@
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++17";
 				CLANG_ENABLE_OBJC_WEAK = YES;
+				"CODE_SIGN_IDENTITY[sdk=macosx*]" = "-";
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 8SXR2327BM;
```
Comment (NachoSoto) [inline RevenueCat.xcodeproj/project.pbxproj:2281]: Oops, did you mean to change these?

**Review by NachoSoto — APPROVED** [review]
Just one comment on the project file.

Thanks for doing this!

File: BackendIntegrationTests/BackendIntegrationTests.swift (hunk 5 above)
Comment (aboedo) [reply BackendIntegrationTests/BackendIntegrationTests.swift:251]: I get an `invalid redeclaration` when trying to reuse the name in these
image

File: RevenueCat.xcodeproj/project.pbxproj (hunk 7 above)
Comment (aboedo) [reply RevenueCat.xcodeproj/project.pbxproj]: reverted 👍

File: Purchases/Error Handling/BackendErrorCode.swift (hunk 6 above)
Comment (aboedo) [reply Purchases/Error Handling/BackendErrorCode.swift:59]: ❤️  it

File: BackendIntegrationTests/BackendIntegrationTests.swift (hunk 5 above)
Comment (aboedo) [reply BackendIntegrationTests/BackendIntegrationTests.swift:251]: @NachoSoto I'm merging these for now, but happy to follow up with another PR if we come up with a good way to handle them

Comment (NachoSoto) [reply BackendIntegrationTests/BackendIntegrationTests.swift:251]: Oh this is fine then!

**Pull request merged**